Lib/multiprocessing/connection.py

```diff
@@ -272,11 +274,24 @@ class PipeConnection(_ConnectionBase):
         """
         _got_empty_message = False
 
+        def __init__(self, *args, **kwargs):
+            super().__init__(*args, **kwargs)
+            self._send_ov = None
+
         def _close(self, _CloseHandle=_winapi.CloseHandle):
+            ov = self._send_ov
+            if ov is not None:
+                # Interrupt WaitForMultipleObjects() in _send_bytes()
+                ov.cancel()
```

Comment:
asyncio uses a similar code in ProactorEventLoop:

https://github.com/python/cpython/blob/1ec45378e97a2d3812ed8bc18e1bd1eb8ddcc9a0/Lib/asyncio/windows_events.py#L67-L81

asyncio uses more advanced code around to handle more cases. For example, in asyncio, the cancel() API is part of the public API.

Here the cancellation is a standard action in the Windows Overlapped API. The cancellation is synchronous, it's easy!

Hopefully, we are not in the very complicated ``RegisterWaitWithQueue()`` case! This case requires an **asynchronous** cancellation which is really complicated to handle: the completion of the cancellation should be awaited!? See this horror story: https://vstinner.github.io/asyncio-proactor-cancellation-from-hell.html